notion-distcopy/src/test/scala/com/ambiata/notion/distcopy/SynchronizedInputsOutputsSpec.scala

```diff
@@ -107,13 +111,47 @@ class SynchronizedInputsOutputsSpec extends AwsScalaCheckSpec(tests = 5) with Di
     val isHdfsNoSync = isHdfsLocation(location) && syncDirIsHdfs
 
     valid1 ==> valid2 :| "v1 ==> v2" &&
-    valid2 ==> valid1 :| "v2 ==> v1" &&
-    (!isHdfsNoSync ||
-      ((location, syncLocation.toOption) must beLike {
-        case (HdfsLocation(p), Some(sl)) => sl.path.toUri.toString ==== p
-      }))
+    valid2 ==> valid1 :| "v2 ==> v1"
   }
 
+  def syncPaths =
+    // local
+    prop { (location: LocalLocation, syncDir: ExecutionLocation) =>
+      val syncLocation = createSynchronizedLocation(Some(syncDir), location).toOption.get
+      val syncDirIsHdfs = syncDir.fold(_ => true, _ => false)
+
+      (syncLocation.path ==== new Path(location.path)).when(!syncDirIsHdfs) and
+      (syncLocation.path ==== new Path(syncDir.path.toUri.getPath, DirPath.unsafe(location.path).asRelative.path)).when(syncDirIsHdfs)
+    } &&
+    prop { location: LocalLocation =>
+      val syncLocation = createSynchronizedLocation(None, location).toOption.get
+      syncLocation.path ==== new Path(location.path)
+    } &&
+    // hdfs
+    prop { (location: HdfsLocation, syncDir: ExecutionLocation) =>
+      val syncLocation = createSynchronizedLocation(Some(syncDir), location).toOption
+      val syncDirIsHdfs = syncDir.fold(_ => true, _ => false)
+
+      (syncLocation ==== None).when(!syncDirIsHdfs) and
+      (syncLocation.map(_.path) ==== Some(new Path(location.path))).when(syncDirIsHdfs)
+    } &&
+    prop { location: HdfsLocation =>
+      val syncLocation = createSynchronizedLocation(None, location).toOption.get
+      syncLocation.path ==== new Path(location.path)
+    } &&
+    // S3
+    prop { (location: S3Location, syncDir: ExecutionLocation) =>
+      val syncLocation = createSynchronizedLocation(Some(syncDir), location).toOption.get
+      val syncDirIsHdfs = syncDir.fold(_ => true, _ => false)
+
+      (syncLocation.path ==== new Path(syncDir.path.toUri.getPath, DirPath.unsafe(location.key).path)).when(!syncDirIsHdfs) and
+      (syncLocation.path ==== new Path(syncDir.path.toUri.getPath, DirPath.unsafe(location.key).path)).when(syncDirIsHdfs)
```

Comment:
Could this be reduced to just `(syncLocation.path ==== new Path(syncDir.path.toUri.getPath, DirPath.unsafe(location.key).path))`?